scrapers/vixenNetwork.py

```diff
@@ -1,126 +1,150 @@
-    if scene.get('images'):
-        scrape['image'] = scene['images']['poster'][len(scene['images']['poster']) - 1]['src']
-    if len(sys.argv) > 1:
-        save_json(j, scene['videoId'], sys.argv[1])
-    print(json.dumps(scrape))
-
-
-if __name__ == '__main__':
-    if sys.version_info.major != MINIMUM_VERSION_MAJOR or sys.version_info.minor < MINIMUM_VERSION_MINOR:
-        log('Invalid Python version. Version %s.%s or later required.' % (MINIMUM_VERSION_MAJOR, MINIMUM_VERSION_MINOR))
-        sys.exit(1)
     try:
-        main()
-    except Exception as e:
-        log(traceback.format_exc())
-        log(e)
+        response = requests.post(self.api, json=j, headers=headers)
+        if response.status_code == 200:
+            result = response.json()
+            if result.get("error"):
+                for error in result["error"]["errors"]:
+                    raise Exception(f"GraphQL error: {error}")
+            return result
+        else:
+           raise ConnectionError(f"GraphQL query failed:{response.status_code} - {response.content}")
+    except Exception as err:
+        log.error(f"GraphqQL query failed {err}")
+        return None
+
+  def parse_scene(self,response):
+        scene = {}
+        if response is None or response.get('data') is None:
+          return scene
+
+        data = response['data'].get('findOneVideo')
+        if data:
+          scene['title'] = data.get('title')
+          scene['details'] = data.get('description')
+          scene['studio'] = {"name": self.name}
+
+          date = data.get('releaseDate')
+          if date:
+            scene['date']=date.split("T")[0]
+          scene['performers'] = []
+          if data.get('models'):
+            for model in data['models']:
+              scene['performers'].append({"name": model['name']})
+
+          scene['tags'] = []
+          if data.get('tags'):
+            for tag in data['tags']:
+                scene['tags'].append({"name": tag})
+
+          if data.get('images') and data['images'].get('poster'):
+            maxWidth = 0
+            for image in data['images']['poster']:
+              if image['width'] > maxWidth:
+                scene['image'] = image['src']
+              maxWidth = image['width']
+
+          return scene
+
+  getVideoQuery = """query getVideo($videoSlug: String, $site: Site) {
+    findOneVideo(input: {slug: $videoSlug, site: $site}) {
+      title
+      description
+      releaseDate
+      models {
+        name
+      }
+      images {
+        poster {
+          src
+          width
+        }
+      }
+      tags
+  }
+  }
+  """
+
+studios = {
+        Site('Blacked'),
+        Site('Blacked Raw'),
+        Site('Deeper'),
+        Site('Tushy'),
+        Site('Tushy Raw'),
+        Site('Slayed'),
+        Site('Vixen')
+}
+
+frag = json.loads(sys.stdin.read())
+url = frag.get("url")
+if url is None:
+  log.error(f"No URL given")
```

Comment:
don't need the `f`

Reply:
Just thinking that the whole block `if url is None:` is useless...
You can't use this scraper if you don't put a URL so this condition will never happens.